### PR TITLE
[MTDSA-324] [DM] Add doc templates for creating and updating sources

### PR DESCRIPTION
**Pull request by danmassie**

### CODE REVIEW

File: app/uk/gov/hmrc/selfassessmentapi/controllers/definition/DocumentationController.scala [hunk 1]
```diff
@@ -34,11 +34,13 @@ trait DocumentationController extends uk.gov.hmrc.api.controllers.DocumentationC
   }
 
   lazy val supportedDocs: Map[String, Xml] = Map(
-    "Create Summary" -> SummaryDocumentation.createSummary,
```
Comment (alexec) [inline app/uk/gov/hmrc/selfassessmentapi/controllers/definition/DocumentationController.scala]: I feel like you could just enumerate these?

**Comment by alexec** [comment]
Look at all the code being deleted here, great stuff!

File: app/uk/gov/hmrc/selfassessmentapi/controllers/definition/DocumentationController.scala (hunk 1 above)
Comment (danmassie) [inline app/uk/gov/hmrc/selfassessmentapi/controllers/definition/DocumentationController.scala]: enumerate in what way?

Comment (alexec) [inline app/uk/gov/hmrc/selfassessmentapi/controllers/definition/DocumentationController.scala]: You could change `Documentation` into a Scala style enum. Annotate each value with the endpoint name (e.g. `@endpointName("Retrieve Summary")` and then just enumerate the values. Let me see if I can do this.

Comment (danmassie) [inline app/uk/gov/hmrc/selfassessmentapi/controllers/definition/DocumentationController.scala]: One step at a time. 

Lets get this merged so everyone else can use it, then look to refactor it. I'm still working on the other non-source docs, so I'll get to any simplification afterwards. I'd rather not introduce annotations, there are other more scala like ways to do it.

Comment (alexec) [inline app/uk/gov/hmrc/selfassessmentapi/controllers/definition/DocumentationController.scala]: I tried this. My idea did not work.

**Pull request merged**